provider/provider_test.go

```diff
@@ -303,3 +305,137 @@ func TestSetConfiguration(t *testing.T) {
 		assert.Equal(t, expected, actual)
 	})
 }
+
+func TestCheck(t *testing.T) {
+	tests := []struct {
+		name string
+		news resource.PropertyMap
+
+		wantErr error
+	}{
+		{
+			name: "can't push a non-canonical image name",
+			news: resource.PropertyMap{
+				"imageName": resource.NewStringProperty("not-fully-qualified-image-name:latest"),
+				"build": resource.NewObjectProperty(
+					resource.PropertyMap{
+						"dockerfile": resource.NewStringProperty("testdata/Dockerfile"),
+					},
+				),
+			},
+			wantErr: reference.ErrNameNotCanonical,
+		},
+		{
+			name: "image name can be non-canonical if not pushing",
+			news: resource.PropertyMap{
+				"imageName": resource.NewStringProperty("not-pushing:latest"),
+				"skipPush":  resource.NewBoolProperty(true),
+				"build": resource.NewObjectProperty(
+					resource.PropertyMap{
+						"dockerfile": resource.NewStringProperty("testdata/Dockerfile"),
+					},
+				),
+			},
+			wantErr: nil,
+		},
+		{
+			name: "image name can be non-canonical if registry server is provided",
+			news: resource.PropertyMap{
+				"imageName": resource.NewStringProperty("foo/bar:latest"),
+				"skipPush":  resource.NewBoolProperty(true),
+				"build": resource.NewObjectProperty(
+					resource.PropertyMap{
+						"dockerfile": resource.NewStringProperty("testdata/Dockerfile"),
+					},
+				),
+				"registry": resource.NewObjectProperty(
+					resource.PropertyMap{
+						"server": resource.NewStringProperty("docker.io"),
+					},
+				),
+			},
+			wantErr: nil,
+		},
+		{
+			name: "image name must be non-canonical if using caching, even when not pushing",
```

Comment:
should this say "image name must be canonical if using caching, even when not pushing" ?

Reply:
Good catch! Fixed.